pkg/progressreader/progressstatus.go

```diff
@@ -0,0 +1,78 @@
+package progressreader
+
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+// ProgressStatus keeps track of one or more observers watching the progress
+// of an operation. For example, if multiple clients are trying to pull an
+// image, they share a ProgressStatus for the download operation.
+type ProgressStatus struct {
```

Comment:
This is more of a broadcaster, than anything else. This isn't really a `ProgressStatus`, which kind of confuses the point of this struct.